### ISSUE TITLE
Compilation Error

### ISSUE THREAD

**Issue by AM4283** [ros-noetic-arch/ros-noetic-moveit-setup-assistant#3]
Hello,
I was receiving this error when trying to install this package as a dependency for ```ros-noetic-moveit```

```
[albi@archlaptop] ➜ ~ yay -S ros-noetic-moveit 
:: Checking for conflicts... 
:: Checking for inner conflicts... 
[Aur:2]  ros-noetic-moveit-setup-assistant-1.1.8-3  ros-noetic-moveit-1.1.8-1 
  2 ros-noetic-moveit-setup-assistant (Build Files Exist) 
  1 ros-noetic-moveit                (Build Files Exist) 
==> Packages to cleanBuild? 
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4) 
==> 
:: PKGBUILD up to date, Skipping (2/0): ros-noetic-moveit-setup-assistant, ros-noetic-moveit 
  2 ros-noetic-moveit-setup-assistant (Build Files Exist) 
  1 ros-noetic-moveit                (Build Files Exist) 
==> Diffs to show? 
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4) 
==> 
:: (1/2) Parsing SRCINFO: ros-noetic-moveit-setup-assistant 
:: (2/2) Parsing SRCINFO: ros-noetic-moveit 
==> Making package: ros-noetic-moveit 1.1.8-1 (Wed 13 Jul 2022 11:51:38 PM EDT) 
==> Retrieving sources... 
  -> Found ros-noetic-moveit-1.1.8.tar.gz 
==> Validating source files with sha256sums... 
    ros-noetic-moveit-1.1.8.tar.gz ... Passed 
==> Making package: ros-noetic-moveit-setup-assistant 1.1.8-3 (Wed 13 Jul 2022 11:51:39 PM EDT) 
==> Retrieving sources... 
  -> Found ros-noetic-moveit-setup-assistant-1.1.8.tar.gz 
==> Validating source files with sha256sums... 
    ros-noetic-moveit-setup-assistant-1.1.8.tar.gz ... Passed 
==> Making package: ros-noetic-moveit-setup-assistant 1.1.8-3 (Wed 13 Jul 2022 11:51:40 PM EDT) 
==> Checking runtime dependencies... 
==> Checking buildtime dependencies... 
==> Retrieving sources... 
  -> Found ros-noetic-moveit-setup-assistant-1.1.8.tar.gz 
==> Validating source files with sha256sums... 
    ros-noetic-moveit-setup-assistant-1.1.8.tar.gz ... Passed 
==> Removing existing $srcdir/ directory... 
==> Extracting sources... 
  -> Extracting ros-noetic-moveit-setup-assistant-1.1.8.tar.gz with bsdtar 
==> Sources are ready. 
==> Making package: ros-noetic-moveit-setup-assistant 1.1.8-3 (Wed 13 Jul 2022 11:51:43 PM EDT) 
==> Checking runtime dependencies... 
==> Checking buildtime dependencies... 
==> WARNING: Using existing $srcdir/ tree 
==> Removing existing $pkgdir/ directory... 
==> Starting build()... 
-- The C compiler identification is GNU 12.1.0 
-- The CXX compiler identification is GNU 12.1.0 
-- Detecting C compiler ABI info 
-- Detecting C compiler ABI info - done 
-- Check for working C compiler: /usr/bin/cc - skipped 
-- Detecting C compile features 
-- Detecting C compile features - done 
-- Detecting CXX compiler ABI info 
-- Detecting CXX compiler ABI info - done 
-- Check for working CXX compiler: /usr/bin/c++ - skipped 
-- Detecting CXX compile features 
-- Detecting CXX compile features - done 
-- Found Boost: /usr/lib64/cmake/Boost-1.79.0/BoostConfig.cmake (found version "1.79.0") found components: thread filesystem system program_options 
-- Using CATKIN_DEVEL_PREFIX: /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/build/devel 
-- Using CMAKE_PREFIX_PATH: /opt/ros/noetic 
-- This workspace overlays: /opt/ros/noetic 
-- Found PythonInterp: /usr/bin/python (found suitable version "3.10.5", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python 
-- Using default Python package layout 
-- Found PY_em: /usr/lib/python3.10/site-packages/em.py 
-- Using empy: /usr/lib/python3.10/site-packages/em.py 
-- Using CATKIN_ENABLE_TESTING: ON 
-- Skip enable_testing() when building binary package 
-- Using CATKIN_TEST_RESULTS_DIR: /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/build/test_results 
-- Forcing gtest/gmock from source, though one was otherwise available. 
-- Found gtest sources under '/usr/src/gmock': gtests will be built 
-- Found gmock sources under '/usr/src/gmock': gmock will be built 
-- Found Python: /usr/bin/python3.10 (found version "3.10.5") found components: Interpreter 
-- Found Threads: TRUE 
-- Using Python nosetests: /usr/bin/nosetests-3.10 
-- catkin 0.8.10 
-- BUILD_SHARED_LIBS is on 
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy 
-- Found ompl: /usr/include/ompl-1.5;/usr/include;/usr/include/eigen3 
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'yaml-cpp>=0.5' 
--   Found yaml-cpp, version 0.7.0 
-- Configuring done 
-- Generating done 
-- Build files have been written to: /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/build 
[  2%] Automatic MOC for target moveit_setup_assistant_tools 
[  2%] Built target moveit_setup_assistant_tools_autogen 
[  7%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/moveit_setup_assistant_tools_autogen/mocs_compilation.cpp.o 
[  7%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/compute_default_collisions.cpp.o 
[ 10%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/moveit_config_data.cpp.o 
[ 12%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/collision_linear_model.cpp.o 
[ 15%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/collision_matrix_model.cpp.o 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/collision_linear_model.cpp: In member function ‘virtual bool SortFilterProxyModel::lessThan(const QModelIndex&, const QModelIndex&) const’: 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/collision_linear_model.cpp:284:34: warning: ‘bool QVariant::operator<(const QVariant&) const’ is deprecated [-Wdeprecated-declarations] 
  284 |     bool smaller = (value_left < value_right); 
      |                                  ^~~~~~~~~~~ 
In file included from /usr/include/qt/QtCore/qabstractitemmodel.h:43, 
                 from /usr/include/qt/QtCore/qabstractproxymodel.h:43, 
                 from /usr/include/qt/QtCore/QAbstractProxyModel:1, 
                 from /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/collision_linear_model.h:39, 
                 from /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/collision_linear_model.cpp:37: 
/usr/include/qt/QtCore/qvariant.h:467:31: note: declared here 
  467 |     QT_DEPRECATED inline bool operator<(const QVariant &v) const 
      |                               ^~~~~~~~ 
[ 17%] Building CXX object CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/rotated_header_view.cpp.o 
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22, 
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23, 
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14, 
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42, 
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25, 
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29, 
                 from /usr/include/boost/shared_ptr.hpp:17, 
                 from /opt/ros/noetic/include/ros/forwards.h:37, 
                 from /opt/ros/noetic/include/ros/common.h:37, 
                 from /opt/ros/noetic/include/ros/ros.h:43, 
                 from /opt/ros/noetic/include/urdf/model.h:50, 
                 from /opt/ros/noetic/include/moveit/robot_model/robot_model.h:42, 
                 from /opt/ros/noetic/include/moveit/planning_scene/planning_scene.h:39, 
                 from /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h:40, 
                 from /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:37: 
/usr/include/boost/bind.hpp:36:1: note: ‘#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.’ 
   36 | BOOST_PRAGMA_MESSAGE( 
      | ^~~~~~~~~~~~~~~~~~~~ 
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22, 
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23, 
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14, 
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42, 
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25, 
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29, 
                 from /usr/include/boost/shared_ptr.hpp:17, 
                 from /opt/ros/noetic/include/ros/forwards.h:37, 
                 from /opt/ros/noetic/include/ros/common.h:37, 
                 from /opt/ros/noetic/include/ros/ros.h:43, 
                 from /opt/ros/noetic/include/urdf/model.h:50, 
                 from /opt/ros/noetic/include/moveit/robot_model/robot_model.h:42, 
                 from /opt/ros/noetic/include/moveit/planning_scene/planning_scene.h:39, 
                 from /home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/compute_default_collisions.cpp:37: 
/usr/include/boost/bind.hpp:36:1: note: ‘#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.’ 
   36 | BOOST_PRAGMA_MESSAGE( 
      | ^~~~~~~~~~~~~~~~~~~~ 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp: In instantiation of ‘bool moveit_setup_assistant::parse(const YAML::Node&, const std::string&, T&, const T&) [with T = std::__cxx11::basic_string<char>; std::string = std::__cxx11::basic_string<char>]’: 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1378:14:   required from here 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1346:8: error: cannot convert ‘const YAML::Node’ to ‘boo ’ in initialization 
 1346 |   bool valid = n; 
      |        ^~~~~ 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp: In instantiation of ‘bool moveit_setup_assistant::parse(const YAML::Node&, const std::string&, T&, const T&) [with T = double; std::string = std::__cxx11::basic_string<char>]’: 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1424:12:   required from here 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1346:8: error: cannot convert ‘const YAML::Node’ to ‘boo ’ in initialization 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp: In instantiation of ‘bool moveit_setup_assistant::parse(const YAML::Node&, const std::string&, T&, const T&) [with T = long int; std::string = std::__cxx11::basic_string<char>]’: 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1809:14:   required from here 
/home/albi/.cache/yay/ros-noetic-moveit-setup-assistant/src/moveit-1.1.8/moveit_setup_assistant/src/tools/moveit_config_data.cpp:1346:8: error: cannot convert ‘const YAML::Node’ to ‘boo ’ in initialization 
make[2]: *** [CMakeFiles/moveit_setup_assistant_tools.dir/build.make:104: CMakeFiles/moveit_setup_assistant_tools.dir/src/tools/moveit_config_data.cpp.o] Error 1 
make[2]: *** Waiting for unfinished jobs.... 
make[1]: *** [CMakeFiles/Makefile2:3080: CMakeFiles/moveit_setup_assistant_tools.dir/all] Error 2 
make: *** [Makefile:136: all] Error 2 
==> ERROR: A failure occurred in build(). 
    Aborting... 
 -> error making: ros-noetic-moveit-setup-assistant 
```

**Comment by AM4283** [comment]
#4 Should fix the issue

**Comment by acxz** [comment]
patched with fef8d6f

**Comment by acxz** [comment]
patch not needed with `1.1.13-1`
